### PR TITLE
Sign the container images we build

### PR DESCRIPTION
**Pull request by JohnGarbutt**
Give the pipeline id-token permissions, so the
action can automatically sign the container images.

As an example, you can verify the signature like this:
```
cosign verify ghcr.io/stackhpc/azimuth-caas-operator:3ec0b58 \
  --certificate-oidc-issuer=https://token.actions.githubusercontent.com \
  --certificate-identity-regexp="https://github.com/stackhpc/azimuth-caas-operator/.github/.*"
```

### CODE REVIEW

File: execution-environment/execution-environment.yml [hunk 1]
```diff
@@ -5,15 +5,15 @@ version: 3
 
 images:
   base_image:
-    name: rockylinux:9-minimal
+    name: registry.access.redhat.com/ubi9/ubi-minimal:latest
```
Comment (JohnGarbutt) [inline execution-environment/execution-environment.yml:8]: This means we get a better look at the CVEs lurking in here.

File: .github/workflows/publish-images.yaml [hunk 2]
```diff
@@ -38,10 +43,15 @@ jobs:
           push: true
           tags: ${{ steps.image-meta.outputs.tags }}
           labels: ${{ steps.image-meta.outputs.labels }}
+          sign_image: true
```
Comment (mkjpryor) [inline .github/workflows/publish-images.yaml:46]: Sign every image? Or maybe only when we build a release?

Comment (JohnGarbutt) [reply .github/workflows/publish-images.yaml:46]: I was thinking we should sign everything. I think its best not to do "special things" on release, unless we have to.

**Review by mkjpryor — APPROVED** [review]
LGTM

**Pull request merged**